tests/bugfixes/github/test_issue_790.py

```diff
@@ -0,0 +1,23 @@
+import system_tests
+
+# Due to an integer overflow bug (#790), this test triggers an 8GB
+# memory allocation.  So the test will fail with a std::bad_alloc
+# exception if less than 8GB is available.  On Linux, you can use
+# `ulimit -v 8000000` to reduce the available memory to slightly less
+# than 8GB.
+
+class integer_overflow_in_PngImage_readMetadata(
```

Comment:
Please use `CamelCase`:
```suggestion
class IntegerOverflowInPngImageReadMetadata(
```